tests/test_executors.py

```diff
@@ -293,6 +294,27 @@ def test_mpexec_failure_failfast(self):
             mpexec.execute(qgraph, butler=None)
         self.assertCountEqual(qexec.getDataIds("detector"), [0])
 
+    def test_mpexec_num_fd(self):
+        """Check that number of open files stays reasonable
+        """
+
+        taskDef = TaskDefMock()
+        qgraph = QuantumGraphMock([
+            QuantumIterDataMock(index=i, taskDef=taskDef, detector=i) for i in range(20)
+        ])
+
+        this_proc = psutil.Process()
+        num_fds_0 = this_proc.num_fds()
+
+        # run in multi-process mode, the order of results is not defined
+        qexec = QuantumExecutorMock(mp=True)
+        mpexec = MPGraphExecutor(numProc=3, timeout=100, quantumExecutor=qexec)
+        mpexec.execute(qgraph, butler=None)
+
+        num_fds_1 = this_proc.num_fds()
+        # they should be the same but allow small growth just in case
+        self.assertLess(num_fds_1 - num_fds_0, 5)
```

Comment:
Can you note somewhere what this test gave as the num_fds without your fix? From this line I can't tell where +5 is reasonable or not.

Reply:
I'll add a comment, without fix it "leaked" 1 file descriptor per sub-process/quantum, totals 20 in this case.